homeassistant/components/fronius/sensor.py

```diff
@@ -695,31 +696,13 @@ class FroniusSensorEntityDescription(SensorEntityDescription):
 ]
 
 
-class _FroniusSensorEntity(CoordinatorEntity["FroniusCoordinatorBase"], SensorEntity):
+class _FroniusSensorEntity(_FroniusEntity, SensorEntity):
     """Defines a Fronius coordinator entity."""
 
     entity_description: FroniusSensorEntityDescription
 
     _attr_has_entity_name = True
 
-    def __init__(
-        self,
-        coordinator: FroniusCoordinatorBase,
-        description: FroniusSensorEntityDescription,
-        solar_net_id: str,
-    ) -> None:
-        """Set up an individual Fronius meter sensor."""
-        super().__init__(coordinator)
-        self.entity_description = description
-        self.response_key = description.response_key or description.key
-        self.solar_net_id = solar_net_id
-        self._attr_native_value = self._get_entity_value()
```

Comment:
I didn't notice when I refactored the class that in the `__init__` it was doing the value set in a way that only works for sensors. This was the same problem that was preventing binary sensors from working at all initially, it needs to store a boolean value in a different instance variable.